plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveUtil.java

```diff
@@ -612,7 +614,12 @@ public static NullableValue parsePartitionValue(String partitionName, String val
             if (isNull) {
                 return NullableValue.asNull(DATE);
             }
-            return NullableValue.of(DATE, datePartitionKey(value, partitionName));
+            try {
+                return NullableValue.of(DATE, datePartitionKey(value, partitionName));
+            }
+            catch (TrinoException e) {
+                return NullableValue.asNull(DATE); // Return null instead of exception for compatibility with Hive
```

Comment:
That looks very weird. Why would we do this for dates and not for any other type?

also, can we drop the change here? i understand extending parsing patters is they key

Reply:
I guess other types also needs fix though I haven't yet investigated. 

Reverting this change will cause the below exception:
```
io.trino.tempto.query.QueryExecutionException: java.sql.SQLException: Query failed (#20220408_144406_00000_av5dg): Invalid partition value '1965-09-10 invalid' for DATE partition key: dt=1965-09-10 invalid
	at io.trino.tempto.query.JdbcQueryExecutor.execute(JdbcQueryExecutor.java:119)
	at io.trino.tempto.query.JdbcQueryExecutor.executeQuery(JdbcQueryExecutor.java:84)
	at io.trino.tests.product.utils.QueryExecutors$1.lambda$executeQuery$0(QueryExecutors.java:55)
	at net.jodah.failsafe.Functions.lambda$get$0(Functions.java:48)
	at net.jodah.failsafe.RetryPolicyExecutor.lambda$supply$0(RetryPolicyExecutor.java:62)
	at net.jodah.failsafe.Execution.executeSync(Execution.java:129)
	at net.jodah.failsafe.FailsafeExecutor.call(FailsafeExecutor.java:376)
	at net.jodah.failsafe.FailsafeExecutor.get(FailsafeExecutor.java:67)
	at io.trino.tests.product.utils.QueryExecutors$1.executeQuery(QueryExecutors.java:55)
	at io.trino.tests.product.hive.TestHiveSparkCompatibility.testReadSparkInvalidDatePartitionName(TestHiveSparkCompatibility.java:485)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
	at org.testng.TestNG.runSuites(TestNG.java:1133)
	at org.testng.TestNG.run(TestNG.java:1104)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
Caused by: java.sql.SQLException: Query failed (#20220408_144406_00000_av5dg): Invalid partition value '1965-09-10 invalid' for DATE partition key: dt=1965-09-10 invalid
	at io.trino.jdbc.AbstractTrinoResultSet.resultsException(AbstractTrinoResultSet.java:1932)
	at io.trino.jdbc.TrinoResultSet.getColumns(TrinoResultSet.java:285)
	at io.trino.jdbc.TrinoResultSet.create(TrinoResultSet.java:61)
	at io.trino.jdbc.TrinoStatement.internalExecute(TrinoStatement.java:262)
	at io.trino.jdbc.TrinoStatement.execute(TrinoStatement.java:240)
	at io.trino.tempto.query.JdbcQueryExecutor.executeQueryNoParams(JdbcQueryExecutor.java:128)
	at io.trino.tempto.query.JdbcQueryExecutor.execute(JdbcQueryExecutor.java:112)
	... 33 more
	Suppressed: java.lang.Exception: Query: SELECT value, CAST(dt AS VARCHAR) FROM hive.default.test_trino_reading_spark_date_type_partitioned_ytuqz1imo1o6
		at io.trino.tempto.query.JdbcQueryExecutor.executeQueryNoParams(JdbcQueryExecutor.java:136)
		... 34 more
Caused by: io.trino.spi.TrinoException: Invalid partition value '1965-09-10 invalid' for DATE partition key: dt=1965-09-10 invalid
	at io.trino.plugin.hive.util.HiveUtil.datePartitionKey(HiveUtil.java:791)
	at io.trino.plugin.hive.util.HiveUtil.parsePartitionValue(HiveUtil.java:617)
	at io.trino.plugin.hive.HivePartitionManager.parsePartition(HivePartitionManager.java:291)
	at io.trino.plugin.hive.HivePartitionManager.parseValuesAndFilterPartition(HivePartitionManager.java:241)
	at io.trino.plugin.hive.HivePartitionManager.lambda$getPartitions$3(HivePartitionManager.java:123)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:294)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:169)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:300)
	at java.base/java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
	at io.trino.plugin.hive.HivePartitionManager.getPartitionsAsList(HivePartitionManager.java:162)
	at io.trino.plugin.hive.HiveMetadata.lambda$getTableProperties$71(HiveMetadata.java:2586)
	at java.base/java.util.Optional.or(Optional.java:318)
	at io.trino.plugin.hive.HiveMetadata.getTableProperties(HiveMetadata.java:2580)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.getTableProperties(ClassLoaderSafeConnectorMetadata.java:844)
	at io.trino.metadata.MetadataManager.getTableProperties(MetadataManager.java:401)
	at io.trino.sql.planner.iterative.rule.DetermineTableScanNodePartitioning.apply(DetermineTableScanNodePartitioning.java:59)
	at io.trino.sql.planner.iterative.rule.DetermineTableScanNodePartitioning.apply(DetermineTableScanNodePartitioning.java:33)
	at io.trino.sql.planner.iterative.IterativeOptimizer.transform(IterativeOptimizer.java:202)
	at io.trino.sql.planner.iterative.IterativeOptimizer.exploreNode(IterativeOptimizer.java:169)
	at io.trino.sql.planner.iterative.IterativeOptimizer.exploreGroup(IterativeOptimizer.java:132)
	at io.trino.sql.planner.iterative.IterativeOptimizer.exploreChildren(IterativeOptimizer.java:251)
	at io.trino.sql.planner.iterative.IterativeOptimizer.exploreGroup(IterativeOptimizer.java:134)
	at io.trino.sql.planner.iterative.IterativeOptimizer.exploreChildren(IterativeOptimizer.java:251)
	at io.trino.sql.planner.iterative.IterativeOptimizer.exploreGroup(IterativeOptimizer.java:134)
	at io.trino.sql.planner.iterative.IterativeOptimizer.optimize(IterativeOptimizer.java:117)
	at io.trino.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:239)
	at io.trino.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:216)
	at io.trino.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:211)
	at io.trino.execution.SqlQueryExecution.doPlanQuery(SqlQueryExecution.java:473)
	at io.trino.execution.SqlQueryExecution.planQuery(SqlQueryExecution.java:454)
	at io.trino.execution.SqlQueryExecution.start(SqlQueryExecution.java:395)
	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:243)
	at io.trino.dispatcher.LocalDispatchQuery.lambda$startExecution$7(LocalDispatchQuery.java:143)
	at io.trino.$gen.Trino_376_5_gc9a7119_dirty____20220408_144158_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

Reply:
> i understand NULL is what hive does

Hive returns `1965-09-10` for `1965-09-10 invalid` as https://github.com/trinodb/trino/pull/11873/files#diff-419272f93e3a06773fb726ce00e5afd498f54d14a4cd54629c5c6d1b5315d238R484

However, starting conservatively looks fine to me. I will update the current implementation.